tests/conftest.py

```diff
@@ -31,6 +31,22 @@ def remove_dependency():
 
 # fixture to create a sample file to be used as file_dep
 @pytest.fixture
+def dependency2(request):
```

Comment:
Copy and paste here is not acceptable. 
Note that removing copy and paste here will NOT increase the complexity in the test itself (this code is on conftest).
Hopefully it will not sound contradictory or arbitrary comparing to my previous review.